lib/server.js

```diff
@@ -123,9 +123,9 @@ module.exports = (config) => {
             // Initialize common data in buildFactory and jobFactory
             if (server.app.buildFactory) {
                 server.app.buildFactory.apiUri = server.info.uri;
-                server.app.buildFactory.tokenGen = (buildId, metadata, scmContext) =>
+                server.app.buildFactory.tokenGen = (buildId, metadata, scmContext, expiresSec) =>
                     server.plugins.auth.generateToken(server.plugins.auth
-                        .generateProfile(buildId, scmContext, ['build'], metadata)
+                        .generateProfile(buildId, scmContext, ['build'], metadata), expiresSec
                     );
             }
             if (server.app.jobFactory) {
```

Comment:
Don't we also need to add for `jobFactory.tokenGen`?

Reply:
It is used in here: https://github.com/screwdriver-cd/executor-queue/blob/master/index.js#L166
So we need to make the changes there as well.

Reply:
In `generalToken` method, argument `expiresSec` can be omitted.
https://github.com/screwdriver-cd/screwdriver/pull/1062/files#diff-cc45542b37e76ced3e366e4aa20f512cR91

If there is no `expiresSec` argument, `DEFAULT_EXPIRES_IN` value will be set. It is 2 hours, same as before.
I think if it is not during build event (like periodic events), it is OK that expires time of JWT is fixed 2 hours.